torchaudio/csrc/sox/effects_chain.cpp

```diff
@@ -283,5 +283,131 @@ int64_t SoxEffectsChain::getOutputSampleRate() {
   return interm_sig_.rate;
 }
 
+#ifdef TORCH_API_INCLUDE_EXTENSION_H
+
+namespace {
+
+/// helper classes for passing file-like object to SoxEffectChain
+struct FileObjInputPriv {
+  sox_format_t* sf;
+  py::object* fileobj;
+  char* buffer;
+  uint64_t buffer_size;
+};
+
+/// Callback function to feed byte string
+int fileobj_input_drain(sox_effect_t* effp, sox_sample_t* obuf, size_t* osamp) {
```

Comment:
You could add brief descriptions to the arguments or link some documentation for their choice